k8s/upgrade/src/bin/upgrade-job/upgrade/utils.rs

```diff
@@ -32,26 +34,41 @@ pub(crate) async fn rebuild_result(
         }
 
         for volume in unhealthy_volumes.iter() {
-            match replica_rebuild_count(volume.clone()).await {
-                0 => {
-                    for _i in 0 .. 11 {
-                        // wait for a minute for any rebuild to start
-                        tokio::time::sleep(Duration::from_secs(60_u64)).await;
-                        let count = replica_rebuild_count(volume.clone()).await;
-                        if count > 0 {
-                            return Ok(RebuildResult {
-                                rebuilding: true,
-                                discarded_volumes: stale_volumes.clone(),
-                            });
+            let node = volume
+                .spec
+                .target
+                .as_ref()
+                .ok_or(
+                    EmptyVolumeSpecTarget {
+                        uuid: volume.spec.uuid.to_string(),
+                    }
+                    .build(),
+                )?
+                .node
+                .as_str();
+
+            if node_name == node {
```

Comment:
Making sure if I understand correctly:

- Let's say we are upgrading node named `node3`. 
- A volume named `vol3` is on `node3` (has target there). `vol3` has replicas on `node3` and `node2`. 
- Now with the node name match check we'll be able to detect if e.g `vol3`'s replica on any other node e.g `node2` is rebuilding. `rebuilding` flag will be set `true`.
- In other scenario, if a volume `vol2` whose target is on e.g `node2` and has a rebuilding replica on `node3`, we won't be detecting that.

Reply:
> A volume named vol3 is on node3 (has target there). vol3 has replicas on node3 and node2.
You mean to say `vol3` has replicas on node1 and node2 ?

> Now with the node name match check we'll be able to detect if e.g vol3's replica on any other node e.g node2 is rebuilding. rebuilding flag will be set true.

Here the `rebuild_result` fetches the list of unhealthy volumes . For each unhealthy volume i check the node i.e. The node where front-end IO will be sent to , is the same as the node which is being upgraded.  If both are same then we find the `replica_rebuild_count` and wait for 10 mis  and all those logic . 


> In other scenario, if a volume vol2 whose target is on e.g node2 and has a rebuilding replica on node3, we won't be detecting that.
Yes

Reply:
We should probably check if there is any Healthy child part of rebuilding volume also in the node being upgraded. Otherwise we will end up Faulting one more child from same volume?